File: izooto/src/main/java/com/izooto/feature/pulseweb/PulseWebChromeClient.java
package com.izooto.feature.pulseweb;

import android.content.Context;
import android.view.View;
import android.webkit.WebChromeClient;
import android.webkit.WebView;
import android.widget.ProgressBar;

import com.izooto.Util;

public class PulseWebChromeClient extends WebChromeClient {
    private final String IZ_CLASSNAME = "PulseWebChromeClient";
    private final Context context;
    private final ProgressBar progressBar;

    protected PulseWebChromeClient(Context context, ProgressBar progressBar) {
        this.context = context;
        this.progressBar = progressBar;
    }

    @Override
    public void onProgressChanged(WebView view, int newProgress) {
        super.onProgressChanged(view, newProgress);
        try {
            progressBar.setVisibility(newProgress == 100 ? View.GONE : View.VISIBLE);
        } catch (Exception e) {
            Util.handleExceptionOnce(context, e.toString(), IZ_CLASSNAME, "onProgressChanged");
        }
    }

}

File: izooto/src/main/java/com/izooto/feature/pulseweb/PulseWebChromeClient.kt
package com.izooto.feature.pulseweb

import android.content.Context
import android.view.View
import android.webkit.ConsoleMessage
import android.webkit.WebChromeClient
import android.webkit.WebView
import android.widget.ProgressBar
import com.izooto.Util

internal class PulseWebChromeClient(
    private val context: Context, private val progressBar: ProgressBar
) : WebChromeClient() {
    override fun onProgressChanged(view: WebView, newProgress: Int) {
        super.onProgressChanged(view, newProgress)
        try {
            progressBar.visibility = if (newProgress == 100) View.GONE else View.VISIBLE
        } catch (e: Exception) {
            Util.handleExceptionOnce(
                context, e.toString(), "PulseWebChromeClient", "onProgressChanged"
            )
        }
    }

    override fun onConsoleMessage(consoleMessage: ConsoleMessage): Boolean {
        return true
    }
}
